AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -222,12 +229,19 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
     func subscribeToItemChanges() {
         serialQueue.async { [weak self] in
             guard let self = self else { return }
+
             self.batchItemsChangedSink = self.dataStorePublisher.publisher
                 .filter { _ in !self.dispatchedModelSyncedEvent.get() }
                 .filter(self.filterByModelName(mutationEvent:))
                 .filter(self.filterByPredicateMatch(mutationEvent:))
                 .handleEvents(receiveOutput: self.onItemChangeDuringSync(mutationEvent:) )
-                .collect(.byTimeOrCount(self.serialQueue, self.itemsChangedPeriodicPublishTimeInSeconds, self.itemsChangedMaxSize))
+                .collect(
+                    .byTimeOrCount(
+                        self.serialQueue,
+                        self.itemsChangedPeriodicPublishTimeInSeconds,
+                        self.shouldBatchDuringSync ? self.itemsChangedMaxSize : 1
```

Comment:
perhaps we can open up an issue on the other DataStore repos to let them verify if they have the same behavior, and if so they can do the similar change here